grakn-engine/src/main/java/ai/grakn/engine/postprocessing/PostProcessingTask.java

```diff
@@ -81,95 +74,49 @@ public boolean start(Consumer<TaskCheckpoint> saveCheckpoint, TaskConfiguration
         LOG.trace("Checking post processing should run: " + (timeElapsed >= maxTimeLapse));
 
         // Only try to run if enough time has passed
-        if(timeElapsed > maxTimeLapse){
-            String keyspace = configuration.json().at(KEYSPACE).asString();
-
-            applyPPToMapEntry(configuration, Schema.BaseType.CASTING.name(), keyspace, PostProcessingTask::runCastingFix);
-            applyPPToMapEntry(configuration, Schema.BaseType.RESOURCE.name(), keyspace, PostProcessingTask::runResourceFix);
-
-            return false;
+        if(timeElapsed >= maxTimeLapse){
+            return super.start(saveCheckpoint, configuration);
         }
 
         return true;
     }
 
     @Override
-    public boolean stop() {
-        throw new UnsupportedOperationException("Post processing cannot be stopped while in progress");
-    }
+    public boolean runGraphMutatingTask(GraknGraph graph, Consumer<TaskCheckpoint> saveCheckpoint, TaskConfiguration configuration) {
+        applyPPToMapEntry(graph, configuration, Schema.BaseType.CASTING.name(), PostProcessingTask::runCastingFix);
+        applyPPToMapEntry(graph, configuration, Schema.BaseType.RESOURCE.name(), PostProcessingTask::runResourceFix);
 
-    @Override
-    public void pause() {
-        throw new UnsupportedOperationException("Post processing cannot be paused");
-    }
+        graph.admin().commitNoLogs();
```

Comment:
`graph.admin().commitNoLogs();` closes the graph, so to do this will we need to refactor the way that `AbstractGraphMutationTask` works? I was working under the design that each task would only be one graph mutation.